libmscore/cmd.cpp

```diff
@@ -2947,6 +2947,209 @@ void Score::cmdSlashRhythm()
             }
       }
 
+//---------------------------------------------------------
+//   cmdRealizeChordSymbols
+///   Realize selected chord symbols into notes on the staff.
+///
+///   If a voicing and duration type are specified, the
+///   harmony voicing settings will be overridden by the
+///   passed parameters. Otherwise, the settings set on the
+///   harmony object will be used.
+//---------------------------------------------------------
+
+void Score::cmdRealizeChordSymbols(bool literal, Voicing voicing, HDuration durationType)
+      {
+      const QList<Element*> elist = selection().elements();
+      for (Element* e : elist) {
+            if (!e->isHarmony())
+                  continue;
+            Harmony* h = toHarmony(e);
+            RealizedHarmony r = h->getRealizedHarmony();
+            Segment* seg = toSegment(h->parent());
+            Fraction duration = r.getActualDuration(durationType);
+            Fraction tick = seg->tick();
+            bool concertPitch = styleB(Sid::concertPitch);
+
+            Chord* chord = new Chord(this); //chord template
+            chord->setTrack(h->track()); //set track so notes have a track to sit on
+
+            //create chord from notes
+            QMap<int, int> notes;
```

Comment:
I feel like at one point you had a class RealizedChord or something like that.  That's no longer a thing?  Should it be?  I'm don't really like it when we have data structures like this that we define & construct on the fly rather than with a named type, although to be fair this would hardly be the only case.

Reply:
Partial nevermind, I see RealizedHarmony now, and it contains a member of this same map type.  Maybe it still deserves a name?  NoteMap just for that?  I don't know.

Reply:
Do you mean like its own object or like a typedef?

Reply:
As long as it looks the same to the call code I can't see that matters.  And I don't think there would be any unique members to the class so probably the typedef is fine.  But not really needed, it just sort of bugs me to have to define the same type in multiple places even if it just something simple like this.